api/nodes/serializers.py

```diff
@@ -1142,11 +1142,19 @@ def create(self, validated_data):
         return self.make_instance_obj(node)
 
 
+class RegistrationSchemaRelationshipField(RelationshipField):
+
+    def to_internal_value(self, registration_schema_id):
+        schema = get_object_or_error(RegistrationSchema, registration_schema_id, self.context['request'])
+        if schema.schema_version != LATEST_SCHEMA_VERSION or not schema.active:
+            raise exceptions.ValidationError('Registration supplement must be an active schema.')
+        return {'registration_schema': schema}
```

Comment:
In my opinion, we are overriding the RelationshipField too often, just to override `to_internal_value`.   I have done this again, because I think the alternative will take a little while to put in place, but suggestions for future work are below.

Because RelationshipField extends `HyperlinkedIdentityField`, [`source` is set to `*`](https://github.com/encode/django-rest-framework/blob/38b3d0109bcddd521f1046402cd441a7972ca5ed/rest_framework/relations.py#L434), which means [`source_attrs` is normally []](https://github.com/encode/django-rest-framework/blob/a628a2dbce8f8f3047d30fe5345f86ae843bcdcc/rest_framework/fields.py#L395).  So if you make the RelationshipField writeable (and many=False), it will error with too many values to unpack [here](https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L498), because source_attrs is empty.  We normally create a field that inherits from RelationshipField like I've done here and override `to_internal_value` to get around this.  An alternative is [overriding `source_attrs` on RelationshipField.bind ](https://github.com/CenterForOpenScience/osf.io/pull/8613/commits/e6e04a3ef87799a28f85f73ffa767a2b3c37a42c#diff-4b396d38a0947821679352d308445c5a
).  This worked fine, but it caused a lot of other tests to fail, where we are overriding `to_internal_value` across several RelationshipField-types.   It was getting out of scope to fix all of these instances (NodeLicenseRelationship, for example). 

I suggest we create a ticket(low priority) to explore my more generalized suggestion and dialing back on all of our RelationshipField overrides.